src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -1531,8 +1531,32 @@ void MacroAssembler::check_klass_subtype_slow_path(Register sub_klass,
 
   br(Assembler::NE, *L_failure);
 
-  // Success.  Cache the super we found and proceed in triumph.
-  str(super_klass, super_cache_addr);
+  // Success. Try to cache the super we found and proceed in triumph.
+
+  uint32_t super_cache_backoff = checked_cast<uint32_t>(SecondarySuperMissBackoff);
+  if (super_cache_backoff > 0) {
+    Label L_skip;
+
+    ldrw(rscratch1, Address(rthread, JavaThread::backoff_secondary_super_miss_offset()));
+    sub(rscratch1, rscratch1, 1);
+    strw(rscratch1, Address(rthread, JavaThread::backoff_secondary_super_miss_offset()));
+
+    cmp(rscratch1, (u1) 0);
+    br(Assembler::GT, L_skip);
```

Comment:
```suggestion
    subsw(rscratch1, rscratch1, 1);
    strw(rscratch1, Address(rthread, JavaThread::backoff_secondary_super_miss_offset()));
    br(Assembler::GT, L_skip);
```

Reply:
So, does `subsw` has the same condition-code semantics as its x86 sibling? That's nice to know! Thanks, I applied both changes. And then I realized we can actually piggyback on a single counter update, and make a denser code. See new commit!